lib/manager/dockerfile/extract.ts

```diff
@@ -9,19 +9,25 @@ export function splitImageParts(currentFrom: string): PackageDependency {
   if (currentFrom.includes('$')) {
     // Check if the variable contains default value, and if so return
     // it as a valid dependency
-    const variableRegex = /^\$\{(.+):-(.+):(.+)@sha256:(.+)\}$/i;
+    const variableRegex =
+      /^\$\{(?<varname>\w[\w\d-_]*):-(?<depName>\w[\w\d\-/]*)(:(?<tag>[\w\d.]+))?(@sha256:(?<digest>[\w\d]+))?\}$/i;
```

Comment:
Is there really a need to "reinvent" this parsing or could we reuse some of our existing code for it?

Reply:
We should also use re2, as recently decided 🤗

Reply:
I thought we already had code to parse tag/sha values

Reply:
@rarkins i think this function is the shared code. 😉

Reply:
Hi  @rarkins I searched for existing code implementing such a regex and no luck. So I guess I have to reinvent it here. To be sure that the regex I use is correct, I checked these pages:

https://github.com/compose-spec/compose-spec/blob/master/spec.md#image
https://github.com/opencontainers/.github/blob/master/docs/docs/introduction/digests.md

It seems my regex is correct, but right now I see a couple of things I can improve.

I'll research re2 and try to rewrite my regex in it.

Reply:
I thought we could reuse the logic in https://github.com/renovatebot/renovate/blob/eadb23729f16de9c4781535313aa377df406b90e/lib/manager/dockerfile/extract.ts#L8

Reply:
Hi @rarkins I tried to reuse this logic, thank you for the hint. All tests passed for me, so could you please review again?